### DISCUSSION TITLE
capsules with parent-child modules

### DISCUSSION THREAD

**Discussion by hadaral** [area17/twill#2123] [Q&A]
Hi everyone ! 
we are using capsules and we have an issue which we have not found a solution yet - 

we have a capsule called Area which have 2 modules into it: 1-Area, 2-City.
each Area hasMany cities :

in Area model:
```php
public function cities(): HasMany
{
    return $this->hasMany(City::class);
}
```

we want to use parent-child modules to prevent column with Area beside column with related Cities, something like this:
![Screenshot from 2023-02-01 13-29-19](https://user-images.githubusercontent.com/33652545/216252311-f7d71be2-7675-425e-92fa-9a727e0e4b64.png)

**The real issue is when we want to click on the related cities to prevent its list - it looks like it looking for separate capsule for city 
(even if change City to AreaCity)**

so my question is - is there an option to do that without creating separate capsule for City?

thank you so much for your time !

**Comment by MDylan** [comment]
Hi,

Which Twill version that you are using?
With 2.12.4 I have no problem with this.

Just follow the [documentation ](https://twill.io/docs/crud-modules/nested-modules.html#parent-child-modules) steps.

**Reply by hadaral** [reply]
Hi ! 
i followed the [documentation ](https://twill.io/docs/crud-modules/nested-modules.html#parent-child-modules) steps, 
im still getting this error 
![image](https://user-images.githubusercontent.com/33652545/217749368-c1da7a88-be90-416a-89cd-53618c257ef6.png)

Both of them : Area, AresCity are in the same capsule . under models i have Area, AreaCity. under repositories i have AreaRepository, AreaCityRepository. and also in controllers .
